File: sslscan.py
import subprocess
import sys

def run_sslscan(target_host):
    try:
        # Run the sslscan command and capture the output
        command = f"sslscan {target_host}"
        result = subprocess.run(command, shell=True, capture_output=True, text=True, check=True)

        # Check if the command was successful
        if result.returncode == 0:
            # Print the output
            print("SSL/TLS scan results for", target_host)
            print(result.stdout)
        else:
            print("Error running sslscan. Exit code:", result.returncode)
            print("Error message:", result.stderr)
    except FileNotFoundError:
        print("sslscan command not found. Make sure sslscan is installed and in your system's PATH.")

if __name__ == "__main__":
    target_host = sys.argv[1]
    run_sslscan(target_host)

File: sslscan.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, char *argv[]) {
    if (argc != 2) {
        fprintf(stderr, "Usage: %s <target_host>\n", argv[0]);
        return 1;
    }

    char command[512];
    snprintf(command, sizeof(command), "sslscan %s", argv[1]);

    FILE *fp = popen(command, "r");
    if (fp == NULL) {
        perror("Error opening pipe");
        return 1;
    }

    char buffer[1024];
    size_t n;

    while ((n = fread(buffer, 1, sizeof(buffer) - 1, fp)) > 0) {
        buffer[n] = '\0';
        printf("%s", buffer);
    }

    int status = pclose(fp);
    if (WIFEXITED(status)) {
        if (WEXITSTATUS(status) == 0) {
            return 0;  // Success
        } else {
            fprintf(stderr, "Error running sslscan. Exit code: %d\n", WEXITSTATUS(status));
            return 1;
        }
    } else {
        fprintf(stderr, "Error: sslscan did not exit properly\n");
        return 1;
    }
}
